packages/authentication/README.md

```diff
@@ -2,45 +2,167 @@
+  }
+}
+```
 
+Next, implement a Strategy provider to map strategy names specified
+in `@authenticate` decorators into Passport Strategy instances.
+
+```ts
+// providers/auth-strategy.ts
+import {
+  inject,
+  Provider,
+  ValueOrPromise,
+} from '@loopback/context';
+import {
+  BindingKeys,
+  AuthenticationMetadata,
+} from '@loopback/authentication';
+
+import {Strategy} from 'passport';
+import {BasicStrategy} from 'passport-http';
+
+export class MyAuthStrategyProvider implements Provider<Strategy> {
+  constructor(
+    @inject(BindingKeys.Authentication.METADATA)
+    private metadata: AuthenticationMetadata,
+  ) {}
+
+  value() : ValueOrPromise<Strategy> {
+    const name = this.metadata.strategy;
+    if (name === 'BasicStrategy') {
+      return new BasicStrategy(this.verify);
+    } else {
+      return Promise.reject(`The strategy ${name} is not available.`);
+    }
+  }
+
+  verify(username: string, password: string, cb: Function) {
+    // find user by name & password
+    // call cb(null, false) when user not found
+    // call cb(null, userProfile) when user is authenticated
+  }
+}
+```
 
- ```ts
-  const strategy = new BasicStrategy(async (username, password) => {
-  return await findUser(username, password);
-  };
- getAuthenticatedUser(strategy, ParsedRequest);
+In order to perform authentication, we need to implement a custom Sequence
+invoking the authentication at the right time during the request handling.
+
+```ts
+// sequence.ts
+import {
+  FindRoute,
+  inject,
+  InvokeMethod,
+  ParsedRequest,
+  Reject
+  Send,
+  ServerResponse,
+  SequenceHandler,
+} from '@loopback/core';
+
+import {
+  AuthenticateFn,
+} from '@loopback/authentication';
+
+class MySequence implements SequenceHandler {
+  constructor(
+    @inject('sequence.actions.findRoute') protected findRoute: FindRoute,
+    @inject('sequence.actions.invokeMethod') protected invoke: InvokeMethod,
+    @inject('sequence.actions.send') protected send: Send,
+    @inject('sequence.actions.reject') protected reject: Reject,
+    @inject('authentication.actions.authenticate')
+    protected authenticateRequest: AuthenticateFn,
+  ) {}
+
+  async handle(req: ParsedRequest, res: ServerResponse) {
+    try {
+      const route = this.findRoute(req);
+
+      // This is the important line added to the default sequence implementation
+      const user = await this.authenticateRequest(req);
+
+      const args = await parseOperationArgs(req, route);
+      const result = await this.invoke(route, args);
+      this.send(res, result);
+    } catch (err) {
+      this.reject(res, req, err);
+    }
+  }
+}
 ```
 
+Finally, put it all together in your application object:
+
+```ts
+import {Application} from '@loopback/core';
+import {AuthenticationComponent, BindingKeys} from '@loopback/authentication';
+import {MyAuthStrategyProvider} from './providers/auth-strategy';
+import {MyController} from './controllers/my-controller';
+import {MySequence} from './my-sequence';
```

Comment:
Good catch, I'll fix to `./sequence`. There is usually only a single sequence class in each app, therefore it makes sense to me to keep this custom sequence in `./sequence`, regardless of how the class is actually called.